source/ota_cbor.c

```diff
@@ -48,7 +69,8 @@
  * @param[out] pBlockId Decoded block id value.
  * @param[out] pBlockSize Decoded block size value.
  * @param[out] pPayload Buffer for the decoded payload.
```

Comment:
so `pPayload` is now preallocated and passed in by user? We should then change its type to `uint8_t *` instead of `uint8_t **`.

Reply:
This will require a change in the otaDataInterface (.decodeFileBlock) touching multiple files. We could have a separate PR for this.